spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/AbstractConnectionFactory.java

```diff
@@ -796,12 +799,14 @@ private void keyReadable(final Selector selector, final long now, final Selectio
 		}
 	}
 
-	private void keyAcceptable(final Selector selector, ServerSocketChannel server, final long now) {
-		try {
-			doAccept(selector, server, now);
-		}
-		catch (Exception ex) {
-			logger.error(ex, "Exception accepting new connection(s)");
+	private void keyAcceptable(final Selector selector, @Nullable ServerSocketChannel server, final long now) {
+		if (server != null) {
```

Comment:
Just nitpicking: I tend to prefer `if (server == null) { return; }` as this reduces the level of indentation